src/json.rs

```diff
@@ -0,0 +1,426 @@
+// Vendored from `smoljson` bef592c5da1c3fe38b2462a8d231b0e0c8a86f80, with explicit permission
+// from the author (`thomcc`). Minimized for cc/simplicity. Modifications and additions made to fit cc's needs.
+
+use std::borrow::Cow;
+
+#[derive(PartialEq, Debug, Clone)]
+pub(crate) enum Token<'s> {
+    Null,
+    Bool(bool),
+    NumU(u64),
+    NumI(i64),
+    NumF(f64),
+    StrBorrow(&'s str),
+    StrOwn(Box<str>),
+    Colon,
+    Comma,
+    ObjectBegin,
+    ObjectEnd,
+    ArrayBegin,
+    ArrayEnd,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct Error(());
```

Comment:
I think you could put a str there to make the error more readable:

```suggestion
pub struct Error(Cow<'static, str>);
```